doc/source.rst

```diff
@@ -181,11 +181,12 @@ File patterns
 File path patterns are used for include and omit, and for combining path
 remapping.  They follow common shell syntax:
 
-- ``*`` matches any number of file name characters, not including the directory
-  separator.
-
 - ``?`` matches a single file name character.
 
+- ``*`` matches any number of file name characters, not including the directory
+  separator.  As a special case, if a pattern starts with ``*/``, it is treated
+  as ``**/``, and if a pattern ends with ``/*``, it is treated as ``/**`.
+
 - ``**`` matches any number of nested directory names, including none.
```

Comment:
Having reviewed other places, I'm starting to get a feeling that things like `x/**z/y` and `s/x**z/y` won't work the way it's implied here either. How about adding more examples to this as well?

Reply:
`x/**z/y` and `s/x**z/y` are refused as invalid.

Reply:
Ah, sounds reasonable. Is there a test for this? Should this be mentioned in a "non-working example"?

Reply:
Should this also clarify something like “must not be combined with any other characters in the same path frament of the match pattern”?

Reply:
Out of curiosity, what happens with `"***/**"`?